### ISSUE TITLE
Listener in supper class does not work

### ISSUE THREAD

**Issue by Simonsator** [SpigotMC/BungeeCord#2023]
If I have a class A which looks like this 
`
import net.md_5.bungee.event.EventHandler;
import net.md_5.bungee.api.plugin.Listener;

public class A implements Listener {
	@EventHandler
public void onSomeEvent(Event pEvent){
System.out.println("An event did happen");
}
}
`
And I have a class B:
`
public class B extends A {

}
`
And I register B like this as a Listener in the main plugin class:
ProxyServer.getInstance().getPluginManager().registerListener(this, new A());

The method onSomeEvent(pEvent: Event) is never called. 

In summary methods of the super class which are annotated with EventHandler are not getting executed.

**Comment by kamcio96** [comment]
Same issue exists in Bukkit API

**Comment by Janmm14** [comment]
Not neccessarily @kamcio96.
That depends on each event definition, whether it has its own HandlerList or not (I think at least its like that).

**Comment by kamcio96** [comment]
@Janmm14 I mean EventBus register only top class, without parents

**Comment by md-5** [comment]
Because that's what Java decided with getDeclaredMethods. And to make things better, getMethods doesn't return non public methods, so that would then break someone else's workflow. The only approach then is to implement logic to climb the class hierarchy which imo is overkill.

This is basically a wontfix for me as I don't see the practical benefit of putting event handlers in a super class, and if you really need them you could just declare them locally and call super.

**Comment by Sxtanna** [comment]
Hi, this issue is still open 4 years later, and it really is a massive break in expected behaviour.

> The only approach then is to implement logic to climb the class hierarchy which imo is overkill.

This is not true, please just do what Bukkit does and use a union of `getMethods` and `getDeclaredMethods`.

This will maintain the current behavior of method visibility in the current class, while also creating parity with the behaviour of the sibling platform.

**Comment by A248** [comment]
As much as I would support this idea, it is unfortunately an API break for anyone defining and using their own hierarchy of events, such as in a plugin API.
